templates/tfengine/components/resources/storage_buckets/main.tf

```diff
@@ -21,15 +21,26 @@ module "{{resourceName . "name"}}" {
   project_id = var.project_id
   location   = "{{get . "storage_location" $.storage_location}}"
 
-  {{- if index . "iam_members"}}
-  iam_members = [
-    {{- range .iam_members}}
+  {{- if has . "lifecycle_rules"}}
+  lifecycle_rules = [
```

Comment:
Why can't you use `hclField` here?

Reply:
Unfortunately hclField does not handle objects very well. 

It does `action {` instead of `action = {`